config/config.go

```diff
@@ -0,0 +1,143 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package config // import "go.opentelemetry.io/contrib/config"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/confmap"
```

Comment:
Looks like both `Unmarshal` functions could instead be validate functions, right? And we can implement the `Validate` methods without `confmap`, so that should be enough to remove the `confmap` dependency

Reply:
Removed the unmarshal methods/